components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2013,7 +2031,7 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
 
         validateAttributesForPagination(offset, limit);
 
-        if ("*".equals(filter)) {
+        if (StringUtils.isBlank(filter)) {
```

Comment:
Don't we need to consider the condition where the filter is equal to `*`

Reply:
As we need to send inboundKey even if the filter equals to *, it's accepted in this function and handled in getFilterDataForDBQuery() function in line 3388.